pkg/machine/provider/platform_windows.go

```diff
@@ -40,3 +44,46 @@ func Get() (vmconfigs.VMProvider, error) {
 		return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 	}
 }
+
+// SupportedProviders returns the providers that are supported on the host operating system
+func SupportedProviders() []define.VMType {
+	return []define.VMType{define.HyperVVirt, define.WSLVirt}
+}
+
+// InstalledProviders returns the supported providers that are installed on the host
+func InstalledProviders() ([]define.VMType, error) {
+	installed := []define.VMType{}
+	if wutil.IsWSLInstalled() {
+		installed = append(installed, define.WSLVirt)
+	}
+
+	var outBuf bytes.Buffer
+	cmd := exec.Command("powershell", "Get-WindowsOptionalFeature", "-FeatureName", "Microsoft-Hyper-V-All", "-Online")
+	cmd.Stdout = &outBuf
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
```

Comment:
this returns horrible errors when it fails, something like `exit status X` whuch is not helpful at all.
Make sure to wrap the error message so we know what failed.